flytekit/annotated/node.py

```diff
@@ -60,6 +63,14 @@ def get_registerable_entity(self) -> SdkNode:
                 metadata=self._metadata,
                 sdk_workflow=self._flyte_entity.get_registerable_entity(),
             )
+        elif isinstance(self._flyte_entity, _workflow_model.BranchNode):
+            self._sdk_node = SdkNode(
+                self._id,
+                upstream_nodes=sdk_nodes,
+                bindings=self._bindings,
+                metadata=self._metadata,
+                sdk_branch=self._flyte_entity,
```

Comment:
What's the type of `self._flyte_entity` here?

Reply:
thats the thing, right now its BranchNode. I want to discuss this and clean this up